core/arch/x86/emit_utils.c

```diff
@@ -1337,12 +1337,12 @@ append_restore_simd_reg(dcontext_t *dcontext, instrlist_t *ilist, bool absolute)
         int i;
         uint opcode = move_mm_reg_opcode(true /*align32*/, true /*align16*/);
         ASSERT(proc_has_feature(FEATURE_SSE));
-        for (i = 0; i < NUM_SIMD_SAVED; i++) {
+        for (i = 0; i < proc_num_simd_saved(); i++) {
             APP(ilist,
                 instr_create_1dst_1src(
                     dcontext, opcode, opnd_create_reg(REG_SAVED_XMM0 + (reg_id_t)i),
                     OPND_DC_FIELD(absolute, dcontext, OPSZ_SAVED_XMM,
-                                  XMM_OFFSET + i * XMM_SAVED_REG_SIZE)));
+                                  XMM_OFFSET + i * MAX_SIMD_SLOT_SIZE)));
```

Comment:
This seems odd to me: to have a hardcoded opcde and operands for xmm, but with a MAX_ constant for the size?  This seems like it could end in an error where MAX_ allows for zmm but this instruction only uses xmm.  It seems like the whole insruction needs to vary, not just this size.  Ditto below.

OK after seeing that this MAX_ is the mcontext field size: how about renaming it so it doesn't sound like it varies.  How about MCXT_SIMD_FIELD_SIZE or sthg?

Reply:
Here I agree. There are many other locations in the code where MAX_NUM_SIMD_SLOTS makes more sense. I wanted to avoid having multiple defines if they refer to the same data structure size. I am looking for a name to capture both the "number of slots" and "MCXT data structure size" semantics. Maybe MCTX_MAX_NUM_SIMD_SLOTS.

Reply:
Done.